tests/monitoring/prometheus_utils.go

```diff
@@ -0,0 +1,182 @@
+package monitoring
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"os/exec"
+	"strings"
+	"time"
+
+	. "github.com/onsi/gomega"
+
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests"
+)
+
+type AlertRequestResult struct {
+	Alerts prometheusv1.AlertsResult `json:"data"`
+	Status string                    `json:"status"`
+}
+
+func getAlerts(cli kubecli.KubevirtClient) ([]prometheusv1.Alert, error) {
+	bodyBytes := PrometheusHTTPRequest(cli, "/alerts")
+
+	var result AlertRequestResult
+	err := json.Unmarshal(bodyBytes, &result)
+	if err != nil {
+		return nil, err
+	}
+
+	if result.Status != "success" {
+		return nil, fmt.Errorf("api request failed. result: %v", result)
+	} else {
+		return result.Alerts.Alerts, nil
+	}
+}
+
+func PrometheusHTTPRequest(cli kubecli.KubevirtClient, endpoint string) []byte {
+
+	monitoringNs := getMonitoringNs(cli)
+	token := getAuthorizationToken(cli, monitoringNs)
+
+	var result []byte
+	var err error
+	if tests.IsOpenShift() {
+		url := getPrometheusURLForOpenShift()
+		resp := doHttpRequest(url, endpoint, token)
+		defer resp.Body.Close()
+		result, err = ioutil.ReadAll(resp.Body)
+		Expect(err).NotTo(HaveOccurred())
+	} else {
+		sourcePort := 4321 + rand.Intn(6000)
+		targetPort := 9090
+		Eventually(func() error {
```

Comment:
Why is this encapsulation into an `Eventually` necessary? Isn't the service there right from the start? If it is, I'd rather expect the test to fail directly in case of error and not after a couple of tries?

Reply:
I thought port-forward is not so stable and it may fail because of some network issues. I though it is worth retrying a couple of times. Doesn't it make sense?

Reply:
Ah, yes, I remember now that this is actually an issue.

Reply:
I was just thinking that the `Eventually` you are using in the `waitForPortforwarding` would be enough, but if it isn't then I'm ok with this.